### ISSUE TITLE
error when run index

### ISSUE THREAD

**Issue by jachenms** [HKUDS/MiniRAG#3]
![image](https://github.com/user-attachments/assets/d3119919-1c4b-4ea4-91a9-c1207c23b151)

appreciated to help me to support for the case

**Comment by TianyuFan0504** [comment]
Hi @jachenms , it seems that  your PyTorch installation or device doesn't support CUDA. Please use the following to verify

```
import torch
print(torch.cuda.is_available())  # Should return True
```

**Comment by jachenms** [comment]
do I need to use GPU ? I have only CPU .

**Comment by TianyuFan0504** [comment]
Hi @jachenms, If you only have CPU and want to try MiniRAG, you can use openai-based API to replace the on-device model. Here is an example (change the code in main.py):

```python

os.environ["OPENAI_API_KEY"] = 'YOUR-OPENAI-KEY'

rag = MiniRAG(
    working_dir = WORKING_DIR,
    # llm_model_func = hf_model_complete,#do not use it
    # llm_model_max_token_size = 200,
    # llm_model_name = LLM_MODEL,
    llm_model_func = gpt_4o_mini_complete,
    embedding_func = EmbeddingFunc(
        embedding_dim=384,
        max_token_size=1000,
        func=lambda texts: hf_embedding(
            texts, 
            tokenizer=AutoTokenizer.from_pretrained(EMBEDDING_MODEL),
            embed_model=AutoModel.from_pretrained(EMBEDDING_MODEL)
        )
    ),
)

```

if you want to use **on-device model** (like phi-3.5-mini or GLM-edge, which is mainly used in MiniRAG), I strongly recommend that use GPU to run them.

**Comment by jachenms** [comment]
![Image](https://github.com/user-attachments/assets/fa85b121-0393-488a-808b-498556af46bd)
the sames issue when replacing your code in main.py

![Image](https://github.com/user-attachments/assets/fb0cda08-1fe0-470a-86de-10bbf24fba8a)

please help to check

**Comment by jachenms** [comment]
do you have wechat ? my wechat : jachenms, I will do live streaming for demo your project.

**Comment by TianyuFan0504** [comment]
Hi @jachenms, maybe you should change your torch version. The torch you use is GPU version, try to use CPU version instead:
(from https://pytorch.org/get-started/locally/)

Try this:
```python
pip3 install torch torchvision torchaudio
```


or, change the main.py as follows to **100% avoid ANY local model**:


```python 

os.environ["OPENAI_API_KEY"] = 'YOUR-OPENAI-KEY'

rag = MiniRAG(
    working_dir = WORKING_DIR,
    # llm_model_func = hf_model_complete,#do not use it
    # llm_model_max_token_size = 200,
    # llm_model_name = LLM_MODEL,
    llm_model_func = gpt_4o_mini_complete,
    # embedding_func = EmbeddingFunc(
    #    embedding_dim=384,
    #    max_token_size=1000,
    #    func=lambda texts: hf_embedding(
    #        texts, 
    #        tokenizer=AutoTokenizer.from_pretrained(EMBEDDING_MODEL),
    #        embed_model=AutoModel.from_pretrained(EMBEDDING_MODEL)
    #    )
    # ),
    embedding_func = openai_embedding,
)



```


Note: remember to replace 'YOUR-OPENAI-KEY' with your own openai key

**Comment by jachenms** [comment]
![Image](https://github.com/user-attachments/assets/dd1616ae-13ac-4323-be7d-f981f5925c9f)
yes, I follow your instruction per your rules
but it failed

**Comment by TianyuFan0504** [comment]
please check your Error message, your **embedding dim mismatch**. 
The openai-embedding (text-embedding3-small) dim is 1536,  but the dim you use is 384. 
Please consider using new working dir.

**Comment by jachenms** [comment]
![Image](https://github.com/user-attachments/assets/102e5c87-d536-451c-a91d-02ca9d634883)
new issue happen, please help to support

**Comment by jachenms** [comment]
![Image](https://github.com/user-attachments/assets/101dbea2-7401-45b0-8a69-1cfc053d504c)do 

have I finish your project?

**Comment by TianyuFan0504** [comment]
Yes @jachenms , now MiniRAG system now has a successful response, congrats!

**Comment by jachenms** [comment]
hi ,

Can I invite you for coming to my live stream for sharing your minirag ?  it is valueble to approach your project.
________________________________
发件人: Tianyu Fan ***@***.***>
发送时间: 2025年1月18日 11:10
收件人: HKUDS/MiniRAG ***@***.***>
抄送: jachenms ***@***.***>; Mention ***@***.***>
主题: Re: [HKUDS/MiniRAG] error when run index (Issue #3)


Yes @jachenms<https://github.com/jachenms> , now MiniRAG system now has a successful response, congrats!

―
